src/wallet/db.cpp

```diff
@@ -554,7 +554,6 @@ bool CDB::Rewrite(CWalletDBWrapper& dbw, const char* pszSkip)
         }
         MilliSleep(100);
     }
-    return false;
```

Comment:
promag's solution is a bit more robust against code being moved around, otoh, if the end becomes reachable and there is no return statement the compiler will warn...  though are we sure we won't get false positives for that warning on some compilers?